aws/resource_aws_instance_test.go

```diff
@@ -4659,3 +4718,262 @@ resource "aws_instance" "test" {
+    volume_size           = "11"
+    volume_type           = "standard"
+    delete_on_termination = true
+  }
+
+  tags {
+    Name    = "test-terraform"
+  }
+}
+`
+
+const testAccInstanceConfigUpdateTemplateSettings = `
+data "aws_ami" "debian_jessie_latest" {
+  most_recent = true
+
+  filter {
+    name   = "name"
+    values = ["debian-jessie-*"]
+  }
+
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+
+  filter {
+    name   = "architecture"
+    values = ["x86_64"]
+  }
+
+  filter {
+    name   = "root-device-type"
+    values = ["ebs"]
+  }
+  
+  owners = ["379101102735"] # Debian
+}
+
+resource "aws_launch_template" "foobar" {
+  name_prefix = "foobar"
+  image_id = "${data.aws_ami.debian_jessie_latest.id}"
+  instance_type = "t2.nano"
+}
+
+resource "aws_instance" "foo" {
+  associate_public_ip_address = true
+  count                       = 1
+
+  launch_template {
+    id = "${aws_launch_template.foobar.id}"
+  }
+
+  root_block_device {
+    volume_size           = "10"
+    volume_type           = "standard"
+    delete_on_termination = true
+  }
+
+  tags {
+    Name    = "test-terraform"
+  }
+}
+`
+
+const testAccInstanceConfigUpdateInstanceTemplateVersion = `
+data "aws_ami" "debian_jessie_latest" {
+  most_recent = true
+
+  filter {
+    name   = "name"
+    values = ["debian-jessie-*"]
+  }
+
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+
+  filter {
+    name   = "architecture"
+    values = ["x86_64"]
+  }
+
+  filter {
+    name   = "root-device-type"
+    values = ["ebs"]
+  }
+  
+  owners = ["379101102735"] # Debian
+}
+
+resource "aws_launch_template" "foobar" {
+  name_prefix = "foobar"
+  image_id = "${data.aws_ami.debian_jessie_latest.id}"
+  instance_type = "t2.nano"
+}
+
+resource "aws_instance" "foo" {
+  associate_public_ip_address = true
+  count                       = 1
+
+  launch_template {
+	id = "${aws_launch_template.foobar.id}"
+	version = "1"
+  }
+
+  root_block_device {
+    volume_size           = "10"
+    volume_type           = "standard"
+    delete_on_termination = true
+  }
+
+  tags {
+    Name    = "test-terraform"
+  }
+}
+`
+
+const testAccInstanceConfigUpdateInstanceTemplateLatest = `
+data "aws_ami" "debian_jessie_latest" {
+  most_recent = true
+
+  filter {
+    name   = "name"
+    values = ["debian-jessie-*"]
+  }
+
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+
+  filter {
+    name   = "architecture"
+    values = ["x86_64"]
+  }
+
+  filter {
+    name   = "root-device-type"
+    values = ["ebs"]
+  }
+  
+  owners = ["379101102735"] # Debian
+}
+
+resource "aws_launch_template" "foobar" {
+  name_prefix = "foobar"
+  image_id = "${data.aws_ami.debian_jessie_latest.id}"
+  instance_type = "t2.nano"
+}
+
+resource "aws_instance" "foo" {
+  associate_public_ip_address = true
+  count                       = 1
+
+  launch_template {
+	id = "${aws_launch_template.foobar.id}"
```

Comment:
Should these be 0.12 syntax?

Reply:
Oh, I've made this PR long before 0.12 was released 😃 . I'll fix it.

Reply:
Although, looks like the whole testing suite for instance resource is in 0.11 syntax, so I guess we can leave it as it is.

Reply:
Anyway, I've updated all examples for launch template to 0.12 syntax.